### PR TITLE
lesson5: mail_room_part3, exception handling and comprehension are in…

### PR DESCRIPTION
**Pull request by april301975**
… send_thank_you_add_donation()

### CODE REVIEW

File: students/Thai_H/lesson5/mail_room_part3.py [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+        #we will use this python list comprehension
+        [each_donor['donation'].append(float(donation_amt)) for each_donor in list_of_donors if donor_name == each_donor['name']]
+
+
+    # New donor
+    else:
+        new_donor_name_amt = {'name':donor_name,'donation':[float(donation_amt)]}
+        list_of_donors.append(new_donor_name_amt)
+    # send thank you letter to the the donor name user just entered
+    send_thank_you_letter(donor_name, donation_amt)
```
Comment (charlesrobison) [inline students/Thai_H/lesson5/mail_room_part3.py]: Adding a new donor is not carrying through when going back to the main menu and seeing the list of donors or creating a report.  Can you see why?

File: students/Thai_H/lesson5/mail_room_part3.py [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+    if is_existing_donor(donor_name):
+
+        #--- instead of using this block ------------------------------------
+        #for each_donor in list_of_donors:
+        #    if donor_name == each_donor['name']:
+        #        each_donor['donation'].append(float(donation_amt))
+        #-----------------------------------------------------------------
+
+        #we will use this python list comprehension
+        [each_donor['donation'].append(float(donation_amt)) for each_donor in list_of_donors if donor_name == each_donor['name']]
```
Comment (charlesrobison) [inline students/Thai_H/lesson5/mail_room_part3.py]: Nice!

**Comment by april301975** [comment]
Yes I see what you mean. The init_donor_list() get called every time the
Main_Menu is called and hence wipes out the new donor/donation amt.
Thank you!

**Comment by april301975** [comment]
I moved it from the main() to the
if __name__ == "__main__":
block

**Pull request merged**